src/audio/kpb.c

```diff
@@ -889,10 +889,11 @@ static int kpb_buffer_data(struct comp_dev *dev,
 		/* Are we stuck in buffering? */
 		current_time = platform_timer_get(timer);
 		if (timeout < current_time) {
-			if (current_time - timeout <= UINT_MAX)
+			timeout = platform_timer_ticks_to_ms(timer, current_time - timeout);
```

Comment:
`ticks` also have a specific meaning in Zephyr and it's confusingly different from how SOF uses the same term. If I'm not mixing things up, I think `cycles` in Zephyr is what we call `ticks` in SOF. So how adding two more functions, using the term "ticks" in the SOF sense help us moving closer to the Zephyr API?